planetiler/src/main/java/com/onthegomap/planetiler/examples/LoliStyle.java

```diff
@@ -0,0 +1,128 @@
+package com.onthegomap.planetiler.examples;
+
+import com.onthegomap.planetiler.FeatureCollector;
+import com.onthegomap.planetiler.FeatureMerge;
+import com.onthegomap.planetiler.Planetiler;
+import com.onthegomap.planetiler.Profile;
+import com.onthegomap.planetiler.VectorTile;
+import com.onthegomap.planetiler.geo.GeometryException;
+import com.onthegomap.planetiler.config.Arguments;
+import com.onthegomap.planetiler.reader.SourceFeature;
+import com.onthegomap.planetiler.reader.osm.OsmElement;
+import com.onthegomap.planetiler.reader.osm.OsmRelationInfo;
+
+import java.nio.file.Path;
+import java.util.List;
+
+public class LoliStyle implements Profile {
+
+  @Override
+  public List<OsmRelationInfo> preprocessOsmRelation(OsmElement.Relation relation) {
+    return null;
+  }
+
+  @Override
+  public void processFeature(SourceFeature sourceFeature, FeatureCollector features) {
+
+    // building layer
+    if (sourceFeature.canBePolygon() && sourceFeature.hasTag("building") && !sourceFeature.hasTag("building", "no")) {
+      features.polygon("building")
+        .setMinZoom(15);
+    }
+
+    // boundary layer
+    if (sourceFeature.canBeLine() && (
+      sourceFeature.hasTag("boundary", "administrative") &&
+      sourceFeature.hasTag("admin_level", "2") &&
+      !sourceFeature.hasTag("maritime", "yes")
+    )) {
+      features.line("boundary")
+        .setMinPixelSize(0)
+        .setMinZoom(0);
+    }
+
+    // water layer
+    if (sourceFeature.canBePolygon() && (
+      sourceFeature.hasTag("natural", "water") ||
+      sourceFeature.hasTag("waterway", 
+        "riverbank",
+        "dock",
+        "canal"
+      ) ||
+      sourceFeature.hasTag("landuse",
+        "reservoir",
+        "basin"
+      ) ||
+      "ocean".equals(sourceFeature.getSource())
+    )) {
+      int minZoom = 0;
+      if ("ocean".equals(sourceFeature.getSource())) {
+        minZoom = 0;
+      }
+      else {
+        if (sourceFeature.hasTag("waterway", "dock", "canal")) {
+          minZoom = 10;
+        }
+        else {
+          minZoom = 4;
+        }
+      }
+      features.polygon("water")
+        .setMinZoom(minZoom);
+    }
+  }
+
+  @Override
+  public List<VectorTile.Feature> postProcessLayerFeatures(String layer, int zoom,
+    List<VectorTile.Feature> items) {
+
+    if ("boundary".equals(layer)) {
+      return FeatureMerge.mergeLineStrings(items,
+        0.5,
+        0.5,
+        4
```

Comment:
By nature these planetiler profiles have a lot of magic constants. Do you think we should extract them?

```suggestion
        COME_CONSTANT3,
        COME_CONSTANT1,
        COME_CONSTANT
```

At least for reviews on Github that would nice nice.

Reply:
Applies to all other constants in this file aswell.

Reply:
I factored them out here. We can see how it goes. Moving all constants out probably makes the code just harder to read. For example, the min zoom for lakes should just be defined where it gets used. Especially when it gets used only in one place... Let's see how it goes.